ratelimits/limiter_test.go

```diff
@@ -2,315 +2,331 @@ package ratelimits
-	_, err := l.Check(context.Background(), NewRegistrationsPerIPAddress, tenZeroZeroOne, -1)
-	test.AssertErrorIs(t, err, ErrInvalidCostForCheck)
-}
-
-func Test_Limiter_Check_limit_no_exist(t *testing.T) {
-	l, _ := newTestLimiter(t)
-	_, err := l.Check(context.Background(), Name(9999), tenZeroZeroOne, 1)
+func Test_Limiter_getLimitNoExist(t *testing.T) {
+	t.Parallel()
+	l, err := NewLimiter(clock.NewFake(), newInmem(), "testdata/working_default.yml", "")
+	test.AssertNotError(t, err, "should not error")
+	_, err = l.getLimit(Name(9999), "")
 	test.AssertError(t, err, "should error")
-}
 
-func Test_Limiter_getLimit_no_exist(t *testing.T) {
-	l, _ := newTestLimiter(t)
-	_, err := l.getLimit(Name(9999), "")
-	test.AssertError(t, err, "should error")
 }
 
-func Test_Limiter_with_defaults(t *testing.T) {
-	l, clk := newTestLimiter(t)
-
-	// Attempt to spend 21 requests (a cost > the limit burst capacity), this
-	// should fail with a specific error.
-	_, err := l.Spend(context.Background(), NewRegistrationsPerIPAddress, tenZeroZeroOne, 21)
-	test.AssertErrorIs(t, err, ErrInvalidCostOverLimit)
-
-	// Attempt to spend all 20 requests, this should succeed.
-	d, err := l.Spend(context.Background(), NewRegistrationsPerIPAddress, tenZeroZeroOne, 20)
-	test.AssertNotError(t, err, "should not error")
-	test.Assert(t, d.Allowed, "should be allowed")
-	test.AssertEquals(t, d.Remaining, int64(0))
-	test.AssertEquals(t, d.ResetIn, time.Second)
-
-	// Attempting to spend 1 more, this should fail.
-	d, err = l.Spend(context.Background(), NewRegistrationsPerIPAddress, tenZeroZeroOne, 1)
-	test.AssertNotError(t, err, "should not error")
-	test.Assert(t, !d.Allowed, "should not be allowed")
-	test.AssertEquals(t, d.Remaining, int64(0))
-	test.AssertEquals(t, d.ResetIn, time.Second)
-
-	// Verify our ResetIn is correct. 1 second == 1000 milliseconds and
-	// 1000/20 = 50 milliseconds per request.
-	test.AssertEquals(t, d.RetryIn, time.Millisecond*50)
-
-	// Wait 50 milliseconds and try again.
-	clk.Add(d.RetryIn)
-
-	// We should be allowed to spend 1 more request.
-	d, err = l.Spend(context.Background(), NewRegistrationsPerIPAddress, tenZeroZeroOne, 1)
-	test.AssertNotError(t, err, "should not error")
-	test.Assert(t, d.Allowed, "should be allowed")
-	test.AssertEquals(t, d.Remaining, int64(0))
-	test.AssertEquals(t, d.ResetIn, time.Second)
-
-	// Wait 1 second for a full bucket reset.
-	clk.Add(d.ResetIn)
-
-	// Quickly spend 20 requests in a row.
-	for i := 0; i < 20; i++ {
-		d, err = l.Spend(context.Background(), NewRegistrationsPerIPAddress, tenZeroZeroOne, 1)
-		test.AssertNotError(t, err, "should not error")
-		test.Assert(t, d.Allowed, "should be allowed")
-		test.AssertEquals(t, d.Remaining, int64(19-i))
+func Test_Limiter_CheckWithLimitNoExist(t *testing.T) {
+	t.Parallel()
+	testCtx, limiters, _, testIP := setup(t)
+	for name, l := range limiters {
+		t.Run(name, func(t *testing.T) {
+			_, err := l.Check(testCtx, Name(9999), testIP, 1)
```

Comment:
I think you can mark each of these sub-tests as t.Parallel() too, for maximum funtimes.

Reply:
Tests for the complete package are taking 0.058s so I'm going to pass on this.